### ISSUE TITLE
XY and XYPic to SVG support

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#450]
[Originally Ticket 1774]

The original XYPic ticket is at the arXMLiv Trac:
http://trac.kwarc.info/arXMLiv/ticket/1400

I am working on first basic support for the 0.8 milestone.

Support for XY has been vigorously and continuously requested by the PlanetMath community and is also one of the top used packages in arXiv.org.

**Comment by dginev** [comment]
The now stale contents of my binding experiment for XY are here:

https://github.com/dginev/LaTeXML/tree/xy-support

An example I believe I had working some years back is:
```tex
\documentclass{article}
\usepackage[all]{xypic}

\begin{document}
% 4 nodes with arrows showing a flow between them
\begin{displaymath}
    \xymatrix{ A \ar[r] & B \ar[d] \\
               D \ar[u] & C \ar[l] }
\end{displaymath}
\end{document}
```

Just bookkeeping here, as there is some renewed interest in making the next steps on xy support.

**Comment by holtzermann17** [comment]
Apparently this is still an issue for PlanetMath content:
https://github.com/planetmath/12_Field_theory_and_polynomials/issues/2

**Comment by dginev** [comment]
Since you write about quick fixes in the other issues, just writing to confirm that this issue has remained open for this long exactly due to its complexity. And can sadly confirm no one has XY support as a focus at the moment of writing.

**Comment by brucemiller** [comment]
On 03/27/2018 01:05 PM, Deyan Ginev wrote:
> And can sadly confirm no one has XY support as a focus 
> at the moment of writing.

FWIW, the xy code heavily leverages TeX's `\halign`
in a way that's quite tricky to emulate, particularly
while keeping all the other tabular, etc working.
It's still of interest, and will get looked into,
but alas not in the short term.

**Comment by kohlhase** [comment]
Dear all, 
is there any progress on this? I would be very interested to convert the stacks project, which uses lots and lots (I counted 4043 `\xymatrix`) of xy. Stacks would be a high-profile target for LaTeXML.

**Comment by brucemiller** [comment]
On 2/19/20 10:21 AM, Michael Kohlhase wrote:
> is there any progress on this? 
Yes & No. Since the xy package relies heavily on
the specific behavior of raw TeX's \halign, it
requires an extensive rewrite of it (and ultimately
all of LaTeXML's alignment infrastructure).  A lot,
perhaps most, of that work has been done (in branch halign).
But a lot remains, including the driver layer.
I hope to return to it soon.

**Comment by kohlhase** [comment]
That is very good to hear. Do you have an ETA in mind? 

But I must say that I am a bit confused by the problem in itself. If I look at xymatrix, then that would the "output type" of running LaTeXML over it. When I hear halign, then I suspect it will produce an `mtable` of some kind? 

But given that `xypic` is an "image format", SVG as output looks more suitable. 
What is your thinking on this?

**Comment by dginev** [comment]
As a motivational update: After the 02.2020 run of arXMLiv, xy.sty continues to be the highest impact missing package, at 4.19% with xypic following it at 0.56%

![image](https://user-images.githubusercontent.com/348975/76688671-60048500-6605-11ea-8367-83319da39342.png)

**Comment by dginev** [comment]
As a second motivational update, I was randomly cherry-picking articles from the "new" page of arXiv today and arrived at an xy-using paper from November 2020:
https://arxiv.org/abs/2011.00025

So I am feeling quite eager to give this package another serious shot in the post-release dev period, maybe alongside with Bruce's halign upgrades.

**Comment by dginev** [comment]
@brucemiller should we close here?

**Comment by brucemiller** [comment]
Yay, finally!! Fixed in #1858 

Thanks for the patience :>  (10 years???)
